### PR TITLE
Integrations Testing

### PR DESCRIPTION
**Pull request by KSGulin**
This PR adds automated testing of sparseml integrations

Integrations:
- [x] Base Class #748
- [x] Yolov5 - #747 
- [x] Image Classification #794
- [x] Transformers #797

### CODE REVIEW

**Review by dbogunowicz — DISMISSED** [review]

**Comment by bfineran** [comment]
@KSGulin `typing.Literal` is not supported on python3.6 we'll need an alternative

File: tests/integrations/base_tester.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    def teardown_check(self):
+        """
+        Check for successful environment cleanup.
+        """
+        pass
+
+    def check_file_creation(self, dir):
+        """
+        Check whether files have been created during the run.
+        TODO: Move to universal fixtures file?
```
Comment (rahul-tuli) [inline tests/integrations/base_tester.py:176]: Nit: remove TODO?

File: tests/integrations/helpers.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import os
+from typing import Dict, List
+
+from pydantic import BaseModel
+
+
+def get_configs_with_cadence(cadence: str, dir_path: str = "."):
+    """
+    Find all config files in the given directory with a matching cadence.
+    :param cadence: string signifying how often to run this test. Possible values are:
```
Comment (rahul-tuli) [inline tests/integrations/helpers.py:25]: Nit new line before Params

**Review by bfineran — DISMISSED** [review]

**Review by bfineran — DISMISSED** [review]

**Review by rahul-tuli — DISMISSED** [review]

File: tests/integrations/base_tester.py (hunk 1 above)
Comment (bfineran) [inline tests/integrations/base_tester.py:176]:
```suggestion
```

File: tests/integrations/helpers.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import glob
+import os
+from typing import Dict, List
+
+from pydantic import BaseModel
+
+
+def get_configs_with_cadence(cadence: str, dir_path: str = "."):
+    """
+    Find all config files in the given directory with a matching cadence.
```
Comment (bfineran) [inline tests/integrations/helpers.py:24]:
```suggestion
    Find all config files in the given directory with a matching cadence.

```

**Review by bfineran — DISMISSED** [review]

**Review by bfineran — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**